components/prism-sas.js

```diff
@@ -10,13 +10,132 @@ Prism.languages.sas = {
 			'punctuation': /;/
 		}
 	},
+	'procSQL': {
+		pattern: /(^|\n)(?:proc\s+sql(?:\s+[\w|=]+)?);([\s\S]+(?:\r?\n|\r)(?=((?:proc\s\w+|quit|run|data(?!\=));))((?:proc\s\w+|quit|run|data(?!\=));)|(?:\r?\n|\r)[\s\S]+;)/im,
+		inside: {
+			'fullStep': {
+				pattern: /(^|\n)(?:proc\s+sql(?:\s+[\w|=]+)?);/i,
+				inside: {
+					'step': {
+						pattern: /(?:proc\s\w+|quit|run|data(?!\=))\b/i,
+						alias: 'keyword'
+					},
+					'argument': {
+						pattern: /\w+(?==)/,
+						alias: 'keyword'
+					},
+					'operator': /=/,
+					'punctuation': /;/,
+					'number': /\b(?:[\da-f]+x(?!\w+)|\d+(?:\.\d+)?(?:e[+-]?\d+)?)/,
+					'string': {
+						pattern: /(["'])(?:\1\1|(?!\1)[\s\S])*\1/,
+						greedy: true
+					}
+				}
+			},
+			'step': {
+				pattern: /(?:proc\s\w+|quit|run|data(?!\=))\b/i,
+				alias: 'keyword'
+			},
+			rest: Prism.languages.sql
+		}
+	},
+	'macrodeclaration': {
+		pattern: /((^|\n)(%macro)((\s)+([\w\s\S]+?))+);/im,
+		inside: {
+			keyword: /%macro/im,
+		}
+	},
+	'macroend': {
+		pattern: /((^|\n)(%mend)((\s)+([\w\s\S]+?))+);/im,
+		inside: {
+			keyword: /%mend/im,
+		}
+	},
+	/*%_zscore(headcir, _lhc, _mhc, _shc, headcz, headcpct, _Fheadcz); */
+	'macro': {
+		pattern: /%_\w+(?=\()/,
+		alias: 'keyword'
+	},
+	'input': {
+		pattern: /(\b)+(?:input(?!\=))\s+(?:\w|\$|\&|\s|\-|\.|\/|\*)+;/im,
+		inside:{
+			'input':{
+				alias: 'keyword',
+				pattern: /^input/i,
+			},
+			'comment': [
+				{
+					pattern: /(^\s*|;\s*)\*.*;/m,
+					lookbehind: true
+				},
+				/\/\*[\s\S]+?\*\//
+			],
+			'number': /\b(?:[\da-f]+x(?!\w+)|\d+(?:\.\d+)?(?:e[+-]?\d+)?)/i,
+		}
+	},
 	'comment': [
 		{
-			pattern: /(^\s*|;\s*)\*.*;/m,
+			pattern: /(^\s*|;\s*)\*(.|[\r\n])*?;/m,
```

Comment:
`(.|[\r\n])*?` Isn't this a single line comment?

Reply:
Once you've started a comment in SAS, it can run multiple lines, regardless of line breaks.

For instance:
`* This is
a long comment;`

That would be interpreted as a single comment.

Reply:
Ahhh, ok. But please use `[\s\S]` instead of `(.|[\r\n])`.

Reply:
Will do.